catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/cache/impl/FileSystemPersistenceProvider.java

```diff
@@ -215,7 +168,60 @@ public void clear() {
     }
   }
 
-  private File getMapStoreFile(String key) {
-    return new File(getMapStorePath() + key + SER);
+  Object loadFromPersistence(String key) {
+    File file = getMapStoreFilePath(key).toFile();
+    if (!file.exists()) {
+      return null;
+    }
+    InputStream inputStream = null;
+    try {
+      inputStream = new FileInputStream(Paths.get(persistencePath.toString(), key + SER).toFile());
```

Comment:
❓ Could we do `persistencePath.resolve(key + SER)` instead?

Reply:
Changed